### PR TITLE
Docker container / setup

### PR DESCRIPTION
**Pull request by ckoopmann**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,6 +46,52 @@ jobs:
...
+      - name: Docker build 
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          load: true
+          push: false
+          tags: ${{ env.TEST_TAG }}
+          build-args: |
+            database_url=${{ secrets.DATABASE_URL }} # This DATABASE_URL is only used for sqlx compilation actual db url is specified at runtime
+      - name: Test Container
```
Comment (ckoopmann) [inline .github/workflows/ci.yml:72]: I added this kind of test command verifying that the built container can actually be run.
Lmk if you think this is overkill / uneeded.

Comment (alextes) [reply .github/workflows/ci.yml:72]: Yea overkill I think. It's quite rare a container doesn't run after it both passed the compilation and test steps. Usually a forgotten env variable and even then you'd have to be lucky that the codepath depending on the var is hit during start, not operation. Additionally, if a container immediately crashes, k8s won't terminate the old version and simply stop retrying after a while.

Buuuuut, as the work is already done, I'd say we keep it in. Small benefit is still some benefit and the maintenance burden of this looks minimal 👍 .

File: src/main.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ mod state;
 async fn main() {
     tracing_subscriber::fmt::init();
     let app = app::get_app().await;
-    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
```
Comment (alextes) [inline src/main.rs:12]: 👍

**Review by alextes — APPROVED** [review]
LGTM

**Pull request merged**